File: Excel/527 Challenge.R
library(tidyverse)
library(readxl)
library(Gmisc)

path = "Excel/527 Sum of Digits in Different Bases.xlsx"
input = read_excel(path, range = "A1:A10")
test = read_excel(path, range = "B1:B10")

convert_to_sum <- function(number) {
  digits <- as.numeric(strsplit(as.character(number), "")[[1]]) %>%
    tibble(num = .) %>%
  mutate(row = nrow(.) + 2 -  row_number()) %>%
  rowwise() %>%
  mutate(converted = Gmisc::baseConvert(num,  target = row, base = 10) %>% as.numeric()) %>%
  ungroup() %>%
  summarise(sum = sum(converted)) %>%
  pull()
  return(digits)
}

result = input %>%
  mutate(conv = map_dbl(Number, convert_to_sum)) %>%
  arrange(conv) %>%
  select(`Answer Expected` = Number)

identical(result, test)
#> [1] TRUE 

File: Excel/500-599/527/527 Challenge.py
import pandas as pd
import numpy as np

path = "527 Sum of Digits in Different Bases.xlsx"
input = pd.read_excel(path, usecols = "A")
test  = pd.read_excel(path, usecols = "B")

def convert_to_sum(number):
    digits = [int(d) for d in str(number)]
    rows = np.arange(len(digits), 0, -1) + 1
    digits = [int(np.base_repr(d, base = r)) for d, r in zip(digits, rows)]
    return sum(digits)

result = input.assign(Sum=input["Number"]\
            .apply(convert_to_sum))\
            .sort_values(by="Sum")\
            .drop("Sum", axis=1)\
            .reset_index(drop=True)

print(result["Number"].equals(test["Answer Expected"])) # True